api/apiroutes.go

```diff
@@ -727,15 +727,21 @@ func (c *appContext) getBlockFeeInfo(w http.ResponseWriter, r *http.Request) {
 }
 
 func (c *appContext) getBlockStakeInfoExtended(w http.ResponseWriter, r *http.Request) {
-	idx := c.getBlockHeightCtx(r)
-	if idx < 0 {
+	hash := c.getBlockHashCtx(r)
```

Comment:
In (`NewAPIRouter`) api/apirouter.go, note that `getBlockStakeInfoExtended` is bound to several URL paths:

    rd.Get("/pos", app.getBlockStakeInfoExtended)

Depending on the full path, different middleware is used to embed the height/hash.

- Under /block/best, it is using the `BlockIndexLatestCtx` middleware, which embeds only the best block index in the request context (no hash).
- Under /block/hash/{blockhash}, it is using `BlockHashPathAndIndexCtx`, which embeds the hash from the URL path, looks up the corresponding height, and embeds that too.
- Under /block/{idx}, it is using `BlockIndexPathCtx`, which embeds the index from the path only.
- Under /block/range/{idx0}/{idx}, it is using `m.BlockIndex0PathCtx` and `m.BlockIndexPathCtx`, which embed both indexes from the path only.  Although this path prefix doesn't bind to `getBlockStakeInfoExtended`.

Basically, the `(*appContext).getWhateverCtx(*http.Request)` type functions try to pull out a value.  The middleware set for different paths decides what/when/how the values get put into those context values.

Reply:
OK. This is still a little confusing, but I think I've got it. There are now two distinct methods for the different contexts, and I think I've assigned them correctly. I did see one weird error

`2018-10-29 12:51:32.745 [ERR] RPCC: GetBlockVerbose(0000000000000000000000000000000000000000000000000000000000000000) failed: -5: Block not found: 0000000000000000000000000000000000000000000000000000000000000000`

but I wasn't able to reproduce it. I noticed that in `apirouter`, some index-only-context routes are calling `apiroutes.getBlockVerbose` which uses `getBlockHashCtx`, so I think that this might be another instance of the same error I made.

Reply:
The genesis (block 0) has no previous block, so it is set to that long string, `0000000000000000000000000000000000000000000000000000000000000000`
https://explorer.dcrdata.org/block/0